# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1502409365 25200
#      Thu Aug 10 16:56:05 2017 -0700
# Node ID 889b8262e66198e0c83f9182a19836957c92fe28
# Parent  95797733d539f10db1810bda86d365057ec05f9c
maint: eliminate useless if (p) before free (p)

* lo-utils.cc (octave_fgets): Eliminate useless if before free, using gnulib's
useless-if-before-free maintainer script.
* oct-rl-hist.c (octave_remove_history, octave_replace_history_entry): Likewise.
* toplev.cc (__builtin_delete): Likewise.

diff --git a/libinterp/corefcn/toplev.cc b/libinterp/corefcn/toplev.cc
--- a/libinterp/corefcn/toplev.cc
+++ b/libinterp/corefcn/toplev.cc
@@ -657,13 +657,12 @@ void *
 }
 
 void
 __builtin_delete (void *ptr)
 {
   if (debug_new_delete)
     std::cerr << "__builtin_delete: " << ptr << std::endl;
 
-  if (ptr)
-    free (ptr);
+  free (ptr);
 }
 
 #endif
diff --git a/liboctave/util/lo-utils.cc b/liboctave/util/lo-utils.cc
--- a/liboctave/util/lo-utils.cc
+++ b/liboctave/util/lo-utils.cc
@@ -168,18 +168,17 @@ octave_fgets (FILE *f, bool& eof)
               buf = nullptr;
             }
 
           break;
         }
     }
   while (retval.empty ());
 
-  if (buf)
-    free (buf);
+  free (buf);
 
   octave_quit ();
 
   return retval;
 }
 
 std::string
 octave_fgetl (FILE *f)
diff --git a/liboctave/util/oct-rl-hist.c b/liboctave/util/oct-rl-hist.c
--- a/liboctave/util/oct-rl-hist.c
+++ b/liboctave/util/oct-rl-hist.c
@@ -197,22 +197,19 @@ octave_history_truncate_file (const char
 }
 
 void
 octave_remove_history (int n)
 {
   HIST_ENTRY *discard = remove_history (n);
 
   if (discard)
-    {
-      if (discard->line)
-        free (discard->line);
+    free (discard->line);
 
-      free (discard);
-    }
+  free (discard);
 }
 
 void
 octave_clear_history (void)
 {
   clear_history ();
 }
 
@@ -304,17 +301,14 @@ octave_history_list (int limit, int numb
 }
 
 void
 octave_replace_history_entry (int which, const char *line)
 {
   HIST_ENTRY *discard = replace_history_entry (which, line, 0);
 
   if (discard)
-    {
-      if (discard->line)
-        free (discard->line);
+    free (discard->line);
 
-      free (discard);
-    }
+  free (discard);
 }
 
 #endif
